packages/twenty-server/src/database/typeorm/core/migrations/1724056827317-addInvitation.ts

```diff
@@ -0,0 +1,51 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddInvitation1724056827317 implements MigrationInterface {
+  name = 'AddInvitation1724056827317';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      'ALTER TABLE core."appToken" ALTER COLUMN "userId" DROP NOT NULL',
+    );
+
+    await queryRunner.query(
+      `ALTER TABLE core."appToken" ADD CONSTRAINT "userIdIsNullWhenTypeIsInvitation" CHECK ("appToken".type != 'INVITATION_TOKEN' OR "appToken"."userId" IS NULL)`,
```

Comment:
shouldn't we create a "draft" user instead of adding this edge case
that's being said, we will probably migration apiKeys to appTokens and userId will be possibly null
Just feels that adding the user.email in the "context" + this edge case might be more complex than creating a draft user

Reply:
If you create a draft user you would still need to create a token, so that wouldn't change the fact we have to remove the constraint. Draft user raises a lot of questions (what happens in someone signs up with that email or it's already created, etc?), that I think we'd rather solve upon login/confirmation. I think it could be misused and pollute the DB.